### DISCUSSION TITLE
Confusion from docs

### DISCUSSION THREAD

**Discussion by nandorojo** [semantic-release/semantic-release#1987] [Q&A] [answered]
Question about the plugins order here: https://semantic-release.gitbook.io/semantic-release/usage/plugins#plugins-declaration-and-execution-order

<img width="860" alt="Screen Shot 2021-06-24 at 5 34 43 PM" src="https://user-images.githubusercontent.com/13172299/123335535-9a501800-d512-11eb-9b7b-348d482a123b.png">

Is this explanation of the steps at the bottom correct? If it's in order, shouldn't Git come at the end after NPM? I might be misreading this, but I don't see how the plugins lead to the behavior below.

I'm still getting familiar with semantic-release, so my apologies if I'm missing something clear.

**Comment by travi** [comment, marked as answer]
the important detail in that list that may be confusing you is the mention of steps. each plugin can implement handlers for each step in the process followed by semantic-release and the order of the steps is documented at the top of that page. 

in this example configuration, the list describes execution of the `verifyConditions`, `prepare`, and `publish` implementations in the npm plugin as well as the `verifyConditions` and `prepare` implementations from the git plugin.

within each step, the declaration order matters, but the step order is driven by the overall semantic-release execution.

**Reply by nandorojo** [reply]
I see, so it goes `execute the <step> from the <plugin>` where `step` is happening in order regardless, and within each step, the `plugin` executes in order of the `plugins` array(?)

**Reply by nandorojo** [reply]
I see the steps at the top of the page, but I didn't really take away that they have an order, since it doesn't say so.

**Reply by nandorojo** [reply]
@travi 

1) thanks for the fast reply, very helpful
2) As long as I include a `plugins` array and don't add the `npm` plugin, then it won't attempt to publish to NPM, right?

I'm just discovering this package and am having a bunch of aha moments for how much better my release process could be, so thanks for taking the time.

**Reply by travi** [reply]
> I see the steps at the top of the page, but I didn't really take away that they have an order, since it doesn't say so.

we would welcome a PR that makes this more clear, if you have ideas while you are learning

> As long as I include a plugins array and don't add the npm plugin, then it won't attempt to publish to NPM, right?

does [this part of the documentation](https://semantic-release.gitbook.io/semantic-release/support/faq#can-i-use-semantic-release-to-publish-non-javascript-packages) cover your use case?

**Reply by nandorojo** [reply]
Yeah I could add a PR for the order, assuming I'm correct that the table is displaying them in the order they occur.

**Reply by nandorojo** [reply]
> does this part of the documentation cover your use case?

I think so...I don't totally understand why I would need to globally install semantic-release in order to not publish to npm. This just means I need to use `npx semantic-release`, correct?

I can see that this achieves it, right?

```
['@semantic-release/npm', { npmPublish: false }]
```

I was led to find that from this sentence in the docs you linked to:

> In addition you will need to configure the semantic-release plugins to disable the @semantic-release/npm plugin which is used by default and use a plugin for your project type.

But the code sample used in the docs didn't have that in it, so maybe it isn't actually necessary(?)

This is in the docs:

```json
{
  "plugins": [
    "@semantic-release/commit-analyzer",
    "@semantic-release/release-notes-generator",
    "@semantic-release/github",
    ["@semantic-release/exec", {
      "prepareCmd" : "set-version ${nextRelease.version}",
      "publishCmd" : "publish-package"
      }]
  ]
}
```

Is that `plugins` array missing `['@semantic-release/npm', { npmPublish: false }]`?

**Reply by nandorojo** [reply]
PR opened: https://github.com/semantic-release/semantic-release/pull/1989
